UKHomeOffice/sgmr-service
# NMSW-34 Replace person form autocompletes with new version & extend tests

## AC

Replace the existing autocomplete component with the AlphaGov version that we use on nmsw

----

## To test

- go to the people page
- add a new person
- start typing in issuing state field
- > country options should be shown
- select a country
- start typing in nationality field
- > country options should be shown
- select a nationaltiy
- refresh the page
- > values should persist
- save person
- click on update for that person
- > issuing state and nationality should be prefilled

----

## Notes

While attempting to identify/fix warnings I refactored our API calls to the try/catch, with the call within this component code rather than using an external function (as we do for other repos)

This code includes a bug for the below scenario, next PR will have a resolution for it, currently discussing with UXD our options
- edit a person
- click in a country autocomplete field
- click out of the field
- > error message thrown
